storage/src/main/java/tech/pegasys/teku/storage/server/VersionedDatabaseFactory.java

```diff
@@ -72,10 +78,30 @@ public VersionedDatabaseFactory(
       final String createDatabaseVersion,
       final long stateStorageFrequency,
       final Eth1Address eth1Address) {
+    this(
+        metricsSystem,
+        dataPath,
+        Optional.empty(),
+        dataStorageMode,
+        createDatabaseVersion,
+        stateStorageFrequency,
+        eth1Address);
+  }
+
+  public VersionedDatabaseFactory(
+      final MetricsSystem metricsSystem,
+      final String dataPath,
+      final Optional<String> maybeArchiveDataPath,
+      final StateStorageMode dataStorageMode,
+      final String createDatabaseVersion,
+      final long stateStorageFrequency,
+      final Eth1Address eth1Address) {
     this.metricsSystem = metricsSystem;
     this.dataDirectory = Paths.get(dataPath).toFile();
     this.dbDirectory = this.dataDirectory.toPath().resolve(DB_PATH).toFile();
-    this.archiveDirectory = this.dataDirectory.toPath().resolve(ARCHIVE_PATH).toFile();
+    this.v5ArchiveDirectory = this.dataDirectory.toPath().resolve(ARCHIVE_PATH).toFile();
+    this.v6ArchiveDirectory =
+        maybeArchiveDataPath.map(p -> Paths.get(p).resolve(ARCHIVE_PATH).toFile());
```

Comment:
We want all the dbs to be in the data directory:
```suggestion
    final String archiveDirPath =  shouldCreateSeparateDbs ? ARCHIVE_PATH : DB_PATH
    this.v6ArchiveDirectory = this.dataDirectory.toPath().resolve(archiveDirPath).toFile();
```

Reply:
@ajsutton commented:
>  The original plan was to be able to put the archive dB in a different dir so it could be on a slower bigger disk.

So the exact idea was to allow a different path for archive directory

Reply:
Right - that makes sense actually :)